spyder/api/decoration.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Copyright © <2013-2016> <Colin Duquesnoy and others, see pyqode/AUTHORS.rst>
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+This module contains the text decoration API.
+Adapted from https://github.com/pyQode/pyqode.core/blob/master/pyqode/core/api/decoration.py
+"""
+from qtpy.QtWidgets import QTextEdit
+from qtpy.QtCore import QObject, Signal, Qt
+from qtpy.QtGui import (QTextCursor, QFont, QPen, QColor, QTextFormat,
+                        QTextCharFormat)
+
+
+class TextDecoration(QTextEdit.ExtraSelection):
+    """
+    Helper class to quickly create a text decoration.
```

Comment:
I think It should be moved to the same place that `widgets/sourcecode/api/decorations.py`